File: 04_BehaviouralDesignPatterns/01_IteratorPattern/cpp/badCode.cpp
#include <iostream>
#include <vector>
#include <string>

class Video {
private:
    std::string title;

public:
    Video(const std::string& title) : title(title) {}

    std::string getTitle() const {
        return title;
    }
};

class YouTubePlaylist {
private:
    std::vector<Video> videos;

public:
    void addVideo(const Video& video) {
        videos.push_back(video);
    }

    std::vector<Video>& getVideos() {
        return videos;
    }
};

int main() {
    YouTubePlaylist playlist;
    playlist.addVideo(Video("Video 1"));
    playlist.addVideo(Video("Video 2"));
    playlist.addVideo(Video("Video 3"));

    // Directly accessing videos without an iterator
    for (const auto& video : playlist.getVideos()) {
        std::cout << video.getTitle() << std::endl;
    }

    return 0;
}

File: 04_BehaviouralDesignPatterns/01_IteratorPattern/badCode.java
import java.util.ArrayList;
import java.util.List;

class Video {
    String title;

    public Video(String title) {
        this.title = title;
    }

    public String getTitle() {
        return title;
    }
}

class YoutTubePlaylist {
    private List<Video> videos = new ArrayList<>();

    public void addVideo(Video video) {
        videos.add(video);
    }

    public List<Video> getVideos() {
        return videos;
    }
}

public class badCode {
    public static void main(String[] args) {
        YoutTubePlaylist playlist = new YoutTubePlaylist();
        playlist.addVideo(new Video("Video 1"));
        playlist.addVideo(new Video("Video 2"));
        playlist.addVideo(new Video("Video 3"));

        // Directly accessing videos without an iterator
        for (Video video : playlist.getVideos()) {
            System.out.println(video.getTitle());
        }
    }
}